### ISSUE TITLE
wrond device name and device id in mqtt brocker

### ISSUE THREAD

**Issue by Shaman2** [TradeFace/tuyagateway#82]
I make one gismo model for all my switch 2gang "switch_2gang"

![1](https://user-images.githubusercontent.com/6141429/91043508-60756300-e61c-11ea-9368-29ae6bad075d.jpg)

next I add two gismo with gismo model "switch_2gang"

![2](https://user-images.githubusercontent.com/6141429/91043726-c6fa8100-e61c-11ea-8e33-320952d46167.jpg)

But mqtt discovery device name as second gang name, not gismo name

![3](https://user-images.githubusercontent.com/6141429/91043883-0de87680-e61d-11ea-8acc-3f9c086e418b.jpg)

and object discovery as dp name, dot giamo gang name.

![4](https://user-images.githubusercontent.com/6141429/91043995-3f614200-e61d-11ea-82c9-882d1c370ea8.jpg)

what I make wrong?

**Comment by TradeFace** [comment]
Yeah that is a bit of a problem. Not sure why that happens. It seems it has to do in which order the data is received in HA. Perhaps the name is reset when you restart HA.

**Comment by Shaman2** [comment]
> 
> 
> Yeah that is a bit of a problem. Not sure why that happens. It seems it has to do in which order the data is received in HA. Perhaps the name is reset when you restart HA.

I found it. When I (after add gismo) edit gismo, nothing to change and press "Save" information in mqtt is changing, Object name is right, but device name is not right. See picture, gismo name "switch_child", but in mqtt device name is "swith_child_2", its wrong. Object names "switch_child_1" and "switch_child_2" is right.
So first error only when send information to mqtt after add gismo. After edit gismo object name send right, but device name send wrong.
![5](https://user-images.githubusercontent.com/6141429/91066732-be18a800-e63a-11ea-95bd-ab94c4129c52.jpg)
.
